Tools/CISupport/ews-build/steps.py

```diff
@@ -4623,3 +4623,108 @@ def doStepIf(self, step):
+        return 'NOBODY (OOPS!)'
+
+
+class AddReviewerToCommitMessage(shell.ShellCommand, AddReviewerMixin):
+    name = 'add-reviewer-to-commit-message'
+    haltOnFailure = True
+
+    def __init__(self, **kwargs):
+        super(AddReviewerToCommitMessage, self).__init__(logEnviron=False, **kwargs)
+
+    def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
+        base_ref = self.getProperty('github.base.ref', DEFAULT_BRANCH)
+        head_ref = self.getProperty('github.head.ref', DEFAULT_BRANCH)
+
+        gmtoffset = int(time.localtime().tm_gmtoff * 100 / (60 * 60))
+        date = f'{int(time.time())} {gmtoffset}'
+        self.command = [
+            'git', 'filter-branch', '-f',
+            '--env-filter', f"GIT_AUTHOR_DATE='{date}';GIT_COMMITTER_DATE='{date}'",
+            '--msg-filter', f'sed "s/NOBODY (OO*PP*S!*)/{self.reviewers()}/g"',
+            f'{head_ref}...{base_ref}',
+        ]
+
+        for key, value in self.environment().items():
+            self.workerEnvironment[key] = value
+
+        return super(AddReviewerToCommitMessage, self).start()
+
+    def getResultSummary(self):
+        if self.results == SKIPPED:
+            return {'step': 'No reviewer defined' if self.getProperty('github.number') else 'Patches have no commit message'}
+        elif self.results == SUCCESS:
+            return {'step': f'Reviewed by {self.reviewers()}'}
+        return {'step': 'Failed to apply reviewers'}
+
+    def doStepIf(self, step):
+        return self.getProperty('github.number') and self.getProperty('reviewers_full_names')
+
+    def hideStepIf(self, results, step):
+        return not self.doStepIf(step)
+
+
+class ValidateCommitMessage(shell.ShellCommand):
+    name = 'validate-commit-message'
+    haltOnFailure = True
+
+    def __init__(self, **kwargs):
+        super(ValidateCommitMessage, self).__init__(logEnviron=False, **kwargs)
+
+    def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
+        base_ref = self.getProperty('github.base.ref', DEFAULT_BRANCH)
+        head_ref = self.getProperty('github.head.ref', DEFAULT_BRANCH)
+        self.command = ['git', 'log', head_ref, f'^{base_ref}']
+
+        self.log_observer = BufferLogObserverClass(wantStderr=True)
+        self.addLogObserver('stdio', self.log_observer)
+
+        return super(ValidateCommitMessage, self).start()
+
+    def getResultSummary(self):
+        if self.results == SKIPPED:
+            return {'step': 'Patches have no commit message'}
+        elif self.results == SUCCESS:
+            return {'step': 'Validated commit message'}
+        return {'step': 'Commit message contains (OOPS!)'}
+
+    def evaluateCommand(self, cmd):
+        rc = super(ValidateCommitMessage, self).evaluateCommand(cmd)
+        if rc != SUCCESS:
+            return rc
+
+        log_text = self.log_observer.getStdout()
+        if '(OOPS!)' in log_text:
+            return FAILURE
+        return SUCCESS
```

Comment:
We should also ensure that 'Reviewed by' or 'Unreviewed' or similar text is present in the commit message.